llarp/path/path_context.cpp

```diff
@@ -172,6 +173,46 @@ namespace llarp
           });
     }
 
+    std::optional<std::weak_ptr<TransitHop>>
+    PathContext::TransitHopByInfo(const TransitHopInfo& info)
+    {
+      // this is ugly as sin
+      auto own = MapGet<
+          SyncTransitMap_t::Lock_t,
+          decltype(m_TransitPaths),
+          PathID_t,
+          std::function<bool(const std::shared_ptr<TransitHop>&)>,
+          std::function<TransitHop*(const std::shared_ptr<TransitHop>&)>,
+          TransitHop*>(
+          m_TransitPaths,
+          info.txID,
+          [info](const auto& hop) -> bool { return hop->info == info; },
+          [](const auto& hop) -> TransitHop* { return hop.get(); });
```

Comment:
... and then this can become:
```C++
auto own = MapGet<SyncTransitMap_t::Lock_t>(
    m_TransitPaths,
    info.txID,
    [info](const auto& hop) { return hop->info == info; },
    [](const auto& hop) { return hop.get(); });
 ```